src/wasm.rs

```diff
@@ -240,15 +243,14 @@ where
     fn store_code(&mut self, creator: Addr, code: Box<dyn Contract<ExecC, QueryC>>) -> u64 {
         let code_base_id = self.code_base.len();
         self.code_base.push(code);
-        let code_id = self.code_data.len() + 1;
-        let checksum =
-            HexBinary::from(Sha256::digest(format!("contract code {}", code_id)).to_vec());
+        let code_id = (self.code_data.len() + 1) as u64;
+        let checksum = self.checksum_generator.checksum(creator.clone(), code_id);
```

Comment:
I would pass the creator by reference. The implementor of the trait can always clone it themself if they need it.

Reply:
Exactly, by default it can be passed without additional cloning.